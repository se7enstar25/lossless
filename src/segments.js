import uuid from 'uuid';

import { generateColor } from './util';

export const createSegment = ({ start, end, name } = {}) => ({
  start,
  end,
  name: name || '',
  color: generateColor(),
  uuid: uuid.v4(),
});

export const createInitialCutSegments = () => [createSegment()];

// Because segments could have undefined start / end
// (meaning extend to start of timeline or end duration)
export function getSegApparentStart(seg) {
  const time = seg.start;
  return time !== undefined ? time : 0;
}

export const getCleanCutSegments = (cs) => cs.map((seg) => ({
  start: seg.start,
  end: seg.end,
  name: seg.name,
}));

export function findSegmentsAtCursor(apparentSegments, currentTime) {
  const indexes = [];
  apparentSegments.forEach((segment, index) => {
    if (segment.start < currentTime && segment.end > currentTime) indexes.push(index);
  });
  return indexes;
}
