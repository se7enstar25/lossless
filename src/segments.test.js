import { convertSegmentsToChapters } from './segments';

it('converts segments to chapters with gaps', () => {
  expect(convertSegmentsToChapters([
    {
      start: 104.612,
      end: 189.053,
      name: 'label 1',
    },
    {
      start: 300.448,
      end: 476.194,
      name: 'label 2',
    },
    {
      start: 567.075,
      end: 704.264,
      name: 'label 3',
    },
    {
      start: 704.264,
      end: 855.455,
      name: 'label 4',
    },
  ])).toMatchSnapshot();
});

it('converts segments to chapters with no gaps', () => {
  expect(convertSegmentsToChapters([
    {
      start: 0,
      end: 2,
      name: 'label 1',
    },
    {
      start: 2,
      end: 3,
      name: 'label 2',
    },
  ])).toMatchSnapshot();
});

it('converts segments to chapters with single long segment', () => {
  expect(convertSegmentsToChapters([
    {
      start: 0,
      end: 1,
      name: 'label 1',
    },
  ])).toMatchSnapshot();
});
