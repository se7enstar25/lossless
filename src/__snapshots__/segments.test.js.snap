// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts segments to chapters with gaps 1`] = `
Array [
  Object {
    "end": 104.612,
    "start": 0,
  },
  Object {
    "end": 189.053,
    "name": "label 1",
    "start": 104.612,
  },
  Object {
    "end": 300.448,
    "start": 189.053,
  },
  Object {
    "end": 476.194,
    "name": "label 2",
    "start": 300.448,
  },
  Object {
    "end": 567.075,
    "start": 476.194,
  },
  Object {
    "end": 704.264,
    "name": "label 3",
    "start": 567.075,
  },
  Object {
    "end": 855.455,
    "name": "label 4",
    "start": 704.264,
  },
]
`;

exports[`converts segments to chapters with no gaps 1`] = `
Array [
  Object {
    "end": 2,
    "name": "label 1",
    "start": 0,
  },
  Object {
    "end": 3,
    "name": "label 2",
    "start": 2,
  },
]
`;

exports[`converts segments to chapters with single long segment 1`] = `
Array [
  Object {
    "end": 1,
    "name": "label 1",
    "start": 0,
  },
]
`;
